Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp

```diff
@@ -919,23 +919,42 @@ webm::Status WebMParser::OnBlockGroupBegin(const webm::ElementMetadata&, webm::A
     if (!action)
         return Status(Status::kNotEnoughMemory);
 
+    // Flush any pending samples, so that the only pending samples after BlockGroup ends
+    // are those from this block group.
+    flushPendingVideoSamples();
```

Comment:
The `VideoTrackData::flushPendingSamples()` only keeps frames that have the same timestamp and to determine the duration of the frame.
At the most in practice it will have only 2 frames. 

Can we detect that this file will have HDR metadata? if so we may have to delay flushing the pending video frames until group block end is seen. 

If we do that however it will also break the search for a keyframe. We'll need to split up frames with different time, so we need to make sure the resulting CMSampleBuffer only ever contains frame that have the same timestamp